configure.win

```diff
@@ -33,117 +24,26 @@ NOT_FOUND="not found"
 	RESULT=`echo $1 | sed -e 's/\//\\\\\\\\/g'`
 }
-# slash_to_eight_bslash c:/tmp/blah
-# echo $RESULT
-# blah=$RESULT
-# echo $blah
-
-# A substitute for cygpath, not present in RTools
-# does not work as function (??): sed: -e expression #1, char 8: unterminated `s' command
-# dos_to_cyg() {
-	# RESULT=`echo $1 | sed -e 's/\\/\//g'`
-# }
 
 double_backslash() {
 	RESULT=`echo $1 | sed -e 's/\\/\\\\/g'`
 }
 
-if [ "$MONO_INSTALL_PATH" = "" ]
-then
-## Find the location of the Mono binary installation on Windows.
-if [ ! -e ${RCLR_SRC_DIR}/get_monosdk.cmd ] 
-then
-	echo "error: batch file not found: ${RCLR_SRC_DIR}/get_monosdk.cmd"
-	exit 1
-fi
-GET_MONO_SDK_CMD=`echo ${RCLR_SRC_DIR}/get_monosdk.cmd | sed -e 's/\//\\\\/g'`
-#  echo "cmd /c $CREATE_LIB_CMD" > blah.txt 
-MONO_INSTALL_PATH=`cmd /c "$GET_MONO_SDK_CMD"`
-if [ "$MONO_INSTALL_PATH" = "$NOT_FOUND" ]
-then
-	echo "warning: Path to the Mono SDK not found by querying the Windows registry. Mono support will be disabled"
-	# echo "error: Path to the Mono SDK not found by querying the Windows registry"
-	# exit 1
-fi
-fi # if [ "$MONO_INSTALL_PATH" = "" ]
-
-
-## Location of the sources of the mono codebase
-# there is no way to detect a potential src installation of the Mono codebase. 
-# Likely to be too advanced for the vast majority - let this be set by the user.
-if [ "$MONO_SRC_PATH" = "" ]
-then
-	MONO_SRC_PATH="c:/path_to_be_set_manually"
-fi
-
 # Give info to the R pkg install log
 echo "R_HOME=$R_HOME"
-if [ "$MONO_INSTALL_PATH" != "$NOT_FOUND" ]
-then
-	echo "MONO_INSTALL_PATH=$MONO_INSTALL_PATH"
-fi
-
-
-## Update the rClr.props file, so that the build process finds the correct R.dll and mono-2.0.dll
-
-# For debugging purposes of this script, you need to set R_HOME
-# R_HOME=c:/PROGRA~1/R/R-30~1.0
-
 slash_to_eight_bslash "$R_HOME" # quotes essential for spaces in paths
 R_HOME_BSLASH=$RESULT
-# echo "R_HOME_BSLASH=$R_HOME_BSLASH"
-
-slash_to_eight_bslash "$MONO_INSTALL_PATH" # quotes essential for spaces in paths
-MONO_INSTALL_PATH_BSLASH=$RESULT
-# echo $MONO_INSTALL_PATH_BSLASH
-
-slash_to_eight_bslash "$MONO_SRC_PATH" # quotes essential for spaces in paths
-MONO_SRC_PATH_BSLASH=$RESULT
-# echo $MONO_SRC_PATH_BSLASH
 
-sed -e "s/@R_INSTALL_PATH@/$R_HOME_BSLASH/" -e "s/@MONO_SRC_PATH@/$MONO_SRC_PATH_BSLASH/" \
-	-e "s/@MONO_INSTALL_PATH@/$MONO_INSTALL_PATH_BSLASH/" $RCLR_PROPS_INFILE > $RCLR_PROPS_FILE
-	
-USE_GCC=0
-if [ "$IGNORE_VISUALSTUDIO" != "" ]
-then
-	USE_GCC=1
-	echo "note: Found the flag IGNORE_VISUALSTUDIO. Build will be made with gcc, Mono and xbuild"
-fi
+## Update the rClr.props file, so that the build process finds the correct R.dll
+sed -e "s/@R_INSTALL_PATH@/$R_HOME_BSLASH/" $RCLR_PROPS_INFILE > $RCLR_PROPS_FILE
 
-VS_COMNTOOLS=${VS120COMNTOOLS}
-# echo "$VS_COMNTOOLS"
-if [ "$VS_COMNTOOLS" = "" ]
-then
-	VS_COMNTOOLS=${VS110COMNTOOLS}
-fi
+#Manually set the path!!!
+VS_COMNTOOLS="C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7"
```

Comment:
Unfortunately, I had to specify the path to vscomntools here

Reply:
I would suggest to read the path from an environment variable (e.g. **VSCOMNTOOLS**) or at least to try the environment variable first and only if it's empty use a hardcoded path.
Otherwise it's bound to VS version (and also to VS edition - e.g. it will not work with VS 2019 or VS 2019 Pro).
Having the environment variable **as the 1st choice**: the caller can always adjust it to the system without need to modify configuration files.

Reply:
P.S. shouldn't it be "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/Tools" ? (with **/Tools** at the end)

Reply:
How is the env variable called? Can you give an example how it is on your machine? I don't have any (I thought it created when VStudio is installed, but in my machine it wasnt).

> P.S. shouldn't it be "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/Tools" ? (with /Tools at the end)

This is how it works on my machine.

Reply:
There is no such predefined variable (it's available only in VSTools command line and is version specific there, e.g. for me it is VS160COMNTOOLS). So let's just **define** the name ourselves, e.g. **VSCOMMONN**).
The compilation intruction would be then: either **set** this variable or adjust the path in the win-config file.
(I would also call it **VSCOMMONN** and not **VSCOMMONNTOOLS** because it's pointint to the common folder and not to the common/tools folder for whatever reason)

Reply:
@Yuri05 Using variable now.